### ISSUE TITLE
angular-devkit/build-angular pulling in types/node as a requirement

### ISSUE THREAD

**Issue by sam-s4s** [angular/angular-cli#13784]
# 🐞 Bug report

### Is this a regression?

Yes, the previous version in which this bug was not present was: 5.2.0
(possibly a later version, but this is what I was running successfully - not sure when the bug was re-introduced)

### Description

"@types/node" is being included in node_modules (again). This unfortunately overrides some default types, including setTimeout / setInterval return of type number, instead making the IDE think the return type is NodeJS.Timeout.

The dependency tree or requires is as follows...
@angular-devkit/build-angular
license-webpack-plugin
@types/webpack-sources
@types/node

This has come up at least once before, and was fixed (the angular package causing @types/node to be imported was fixed).

## 🔬 Minimal Reproduction
<pre><code>
let test: number = setTimeout(() => { }, 1000);
</code></pre>

## 🔥 Exception or Error
<pre><code>
(TS) Type 'Timeout' is nto assignable to type 'number'.
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 7.3.3
Node: 10.15.1
OS: win32 x64
Angular: 7.2.6
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.13.3
@angular-devkit/build-angular     0.13.3
@angular-devkit/build-optimizer   0.13.3
@angular-devkit/build-webpack     0.13.3
@angular-devkit/core              7.3.3
@angular-devkit/schematics        7.3.3
@angular/cli                      7.3.3
@ngtools/webpack                  7.3.3
@schematics/angular               7.3.3
@schematics/update                0.13.3
rxjs                              6.4.0
typescript                        3.2.4
webpack                           4.29.0
</code></pre>

**Comment by sam-s4s** [comment]
Just to clarify, as I am doing browser development, types for node should never be included (as they override the default types for browsers). I'm having to type my time handles as `any` to get around this bug.

**Comment by sam-s4s** [comment]
Hmm seems this is being ignored... I thought there would be lots of people using setTimeout and storing a handle as a number... maybe I'm the only one lol

Anyway, I would love this to be fixed :D

No way the node types should be included for Angular deps... it's broken.

**Comment by alan-agius4** [comment]
This issue only effects the IDE. The root cause is that the root level tsconfig includes all the `@types/` which causes the `@types/node` to be included as part of the TypeScript program used by the language server.

We did try to solve this issue using solution style tsconfig, however that created several other issues. See: https://github.com/microsoft/TypeScript/issues/39632

Closing, as this is not actionable from our-end until the above is resolved.
